dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,652 @@
+import java.security.NoSuchAlgorithmException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.apache.commons.lang3.mutable.MutableObject;
+
+/**
+ * Represents a Storage on the database It supports big files since provides the ability to split
+ * fat object in smaller pieces, see {@link FileByteSplitter} and {@link com.dotcms.util.FileJoiner}
+ * to get more details about the process
+ *
+ * @author jsanca
+ */
+public class DataBaseStoragePersistenceAPIImpl implements StoragePersistenceAPI {
+
+    private static final String DATABASE_STORAGE_JDBC_POOL_NAME = "DATABASE_STORAGE_JDBC_POOL_NAME";
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    /**
+     * custom external connection provider method in case we want to store stuff outside our db
+     */
+    protected Connection getConnection() {
+
+        final String jdbcPool = Config
+                .getStringProperty(DATABASE_STORAGE_JDBC_POOL_NAME, StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        return isExternalPool ?
+                DbConnectionFactory.getConnection(jdbcPool) :
+                DbConnectionFactory.getConnection();
+    }
+
+    /**
+     * custom transaction wrapper
+     */
+    protected <T> T wrapInTransaction(final ReturnableDelegate<T> delegate)
+            throws DotDataException {
+
+        final String jdbcPool = Config
+                .getStringProperty(DATABASE_STORAGE_JDBC_POOL_NAME, StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+            return this.wrapInExternalTransaction(delegate, jdbcPool);
+        } else {
+            try {
+                return LocalTransaction.wrapReturnWithListeners(delegate);
+            } catch (Exception e) {
+                Logger.error(DataBaseStoragePersistenceAPIImpl.class, e.getMessage(), e);
+                throw new DotDataException(e);
+            }
+        }
+    }
+
+    /**
+     * custom transaction wrapper
+     */
+    private <T> T wrapInExternalTransaction(final ReturnableDelegate<T> delegate,
+            final String jdbcPool) {
+
+        T result = null;
+        Connection connection = null;
+        boolean autocommit = false;
+
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            autocommit = connection.getAutoCommit();
+            connection.setAutoCommit(false);
+
+            result = delegate.execute();
+
+            connection.commit();
+        } catch (Throwable e) {
```

Comment:
Codacy found an issue: [A catch statement should never catch throwable since it includes errors.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)